xbmc/platform/android/CPUInfoAndroid.cpp

```diff
@@ -31,3 +59,47 @@ CCPUInfoAndroid::CCPUInfoAndroid()
   if (CAndroidFeatures::HasNeon())
     m_cpuFeatures |= CPU_FEATURE_NEON;
 }
+
+float CCPUInfoAndroid::GetCPUFrequency()
+{
+  std::string buffer(64, 0);
+
+  if (ReadFile("/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq", buffer))
```

Comment:
This is the max frequency not the actual frequency (in case of dynamic frequency adjustment).

Also you should just use `CSysfsPath`

Reply:
> This is the max frequency not the actual frequency

I know but seems in line with is used for Windows (3600 MHz is max/nominal speed not actual):

![Captura de pantalla 2021-12-08 173420](https://user-images.githubusercontent.com/58434170/145246832-1b21a3dd-3582-4df7-aeec-51171396a150.png)

This data is not refreshed and it does not make much sense to show something that is not constant value. If it refreshed automatically while staying on the same screen (same as memory usage) then it would be nice to see how it changes.

> Also you should just use CSysfsPath

It uses `std::ifstream` to read files and this c++ seems have troubles in Android due some NDK limitation:
https://stackoverflow.com/questions/32826175/ftello-and-fseeko-android-build-errors

Seems is need `__ANDROID_API__ >= 24` to use this libraries. Because this I used very basic C implementation for file read.

Reply:
> This data is not refreshed and it does not make much sense to show something that is not constant value. If it refreshed automatically while staying on the same screen (same as memory usage) then it would be nice to see how it changes.
> 

Are you sure?

> 
> > Also you should just use CSysfsPath
> 
> 
> 
> It uses `std::ifstream` to read files and this c++ seems have troubles in Android due some NDK limitation:
> 
> https://stackoverflow.com/questions/32826175/ftello-and-fseeko-android-build-errors
> 
> 
> 
> Seems is need `__ANDROID_API__ >= 24` to use this libraries. Because this I used very basic C implementation for file read.
> 
> 

That's fine then. What about using `CPosixFile` or just `CFile` then?